### ISSUE TITLE
Failed test 'CalcColumnWidths - uneven' (with -Duselongdouble?)

### ISSUE THREAD

**Issue by eserte** [PhilterPaper/PDF-Table#60]
On some of my smokers the test suite fails:
```
#   Failed test 'CalcColumnWidths - uneven'
#   at t/PDF-Table.t line 29.
#     Structures begin differing at:
#          $got->[1] = '300.758439703998205'
#     $expected->[1] = '300.758439703998'
# Looks like you failed 1 test of 11.
t/PDF-Table.t .. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/11 subtests 
```

This seems to happen if perl is compiled with increased floating point accuracy (configuration option -Duselongdouble).

**Comment by PhilterPaper** [comment]
I see a handful of these have failed with _slightly_ different floating point values. I'm not sure how I would fix this over the entire product -- presumably _any_ floating point number could come out a little different and break the tests somewhere else. I'm reluctant to round all my floating point results to some small number of decimal places, as that would be a tremendous overhead, as compared to just using the raw results. Any idea what the standard way of handling this is? Wouldn't the use of long doubles (128 bits, I'm guessing) be something that is set by the calling program (here, the .t programs) and not system-wide?

**Comment by eserte** [comment]
> I see a handful of these have failed with slightly different floating point values. I'm not sure how I would fix this over the entire product -- presumably any floating point number could come out a little different and break the tests somewhere else. I'm reluctant to round all my floating point results to some small number of decimal places, as that would be a tremendous overhead, as compared to just using the raw results. Any idea what the standard way of handling this is?

Maybe using something a test module like https://metacpan.org/pod/Test::Deep#num could help. Currently it seems that only one place is affected. Or you could detect such a perl interpreter (using `use Config; if ($Config{uselongdouble} eq "define") { ... }`) and skip parts of the test.

> Wouldn't the use of long doubles (128 bits, I'm guessing) be something that is set by the calling program (here, the .t programs) and not system-wide?

If `-Duselongdouble` was configured when compiling a perl interpreter, then every internal NV is a long double. Currently there's no way to use normal doubles in such interpreters (a pragma like this would be handy in this situation, but probably hard to implement in perl).

**Comment by PhilterPaper** [comment]
Enclosed [PDF-Table.t.txt](https://github.com/PhilterPaper/PDF-Table/files/5491315/PDF-Table.t.txt) is a revised PDF-Table.t test (remove the .txt suffix). It uses the "Config" test you suggested, and works on my normal Strawberry Perl; could you do me a big favor and try running it on a "long double" system? It should not require an actual PDF::API2 or PDF::Builder install, but may require you to grab some lib/ stuff to fake the installation. If you can't, I can go ahead and build PDF-Table-1.001.tar.gz and make it available to you in some manner (I don't want to publish it to CPAN until it's tested). It would also have the revised t/pod.t and t/manifest.t, per #61. Or, if you can easily grab PDF-Table-1.000.tar.gz, unpack it into a directory, replace t/PDF-Table.t with this one, and `perl t/PDF-Table.t`, it might be just as easy. Thanks!

Looking over the failed test results, other than the pod.t complaint about `=head5` and the "long double" on PDF-Table.t (one place), do you know if there were any other "long double" problems in the tests? It's a pity that the test results don't consolidate all the types of problems found, or can they and I missed it? Is there a reasonable way to search all the tests for each type of error found?

**Comment by PhilterPaper** [comment]
I changed things a bit (removed `=head5` from POD, allowed long double in t-tests, made pod.t and manifest.t "author testing") and released 1.001 last night. It appears to be allowing long double Perl now (as well as regular doubles), so I will close this ticket. No further request to test PDF-Table.t.
